### DISCUSSION TITLE
How to refresh the table after adding a new item using modal

### DISCUSSION THREAD

**Discussion by nani1008** [TanStack/query#2719] [Ideas]
How can I refresh the table(react-table) if I add new data and automatically display the newly  added data? So basically on my `index.js` Here is where my `fetchAllServices` query is and I have a modal where I can add a new `service` after adding a new service I want to display it instantly on my react table, tried adding this on my `onSuccess` but still not working my data from the `useQuery(fetchAllServices` is returning `undefined`
```
 onSuccess: () => {
      queryClient.setQueryData("services");
  },
```

**Comment by TkDodo** [comment]
Need a bit more info here please, for example, a codesandbox reproduction that shows your issue.
